specifyweb/frontend/js_src/lib/components/QueryBuilder/reducer.ts

```diff
@@ -52,6 +53,7 @@ export const getInitialState = ({
 }): MainState => ({
   type: 'MainState',
   fields: parseQueryFields(query.fields ?? []),
+  showMappingView: getCache('queryBuilder', 'showMappingView') ?? true,
```

Comment:
would it be possible to call `useCachedState` in the components that need this cache instead and remove `showMappingView` from the MainState?

putting getCache/setState in a state is a bad pattern I used when I was designing first react components in Specify, but don't do that anymore. reasons: central state is more verbose (requires more boilerplate code), leads to worse performance (any change to the state triggers re-render of all components that use the state in any way) and is bug prone (useCachedState listens for cache being updated by some other component somewhere else. getCache/setCache does not listen, thus the state could get out of date)

Reply:
I pushed a code using cachedState but not sure that is what you meant....